### PR TITLE
feat(inputs.prometheus): add support for Custom header

### PR DESCRIPTION
**Pull request by cnanfackgoto**

### CODE REVIEW

File: plugins/inputs/prometheus/prometheus.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ type Prometheus struct {
 	// Basic authentication credentials
 	Username string `toml:"username"`
 	Password string `toml:"password"`
+	
+	ExtraHttpHeaders map[string]string `toml:extra_http_headers`
```
Comment (powersj) [inline plugins/inputs/prometheus/prometheus.go:61]:
```suggestion
	HTTPHeaders map[string]string `toml:http_headers`
```

File: plugins/inputs/prometheus/prometheus.go [hunk 2]
```diff
@@ -296,6 +298,12 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		req.Header.Set("Authorization", "Bearer "+p.BearerTokenString)
 	} else if p.Username != "" || p.Password != "" {
 		req.SetBasicAuth(p.Username, p.Password)
+	} 
+	
+	if p.ExtraHttpHeaders != nil {
+		for key, value := range p.ExtraHttpHeaders {
+ 		 req.Header.Add(fmt.Sprint(key), fmt.Sprint(value))
```
Comment (powersj) [inline plugins/inputs/prometheus/prometheus.go:305]:
```suggestion
	if p.HTTPHeaders != nil {
		for key, value := range p.HTTPHeaders {
 		    req.Header.Add(key, value)                  
```

File: plugins/inputs/prometheus/sample.conf [hunk 3]
```diff
@@ -81,6 +81,10 @@
   # username = ""
   # password = ""
   
+  ## Add extra http headers.
+  # extra_http_headers = [{"header1",  "value1" },{ "header2", "value2" }]
```
Comment (powersj) [inline plugins/inputs/prometheus/sample.conf:85]:
```suggestion
  ## Add custom headers to the HTTP request
  # http_headers = [{"header1",  "value1" },{ "header2", "value2" }]
```

File: plugins/inputs/prometheus/sample.conf [hunk 4]
```diff
@@ -81,6 +81,10 @@
   # username = ""
   # password = ""
   
+  ## Add custom headers to the HTTP request
+  # http_headers = [{"header1",  "value1" },{ "header2", "value2" }]
+
+
```
Comment (powersj) [inline plugins/inputs/prometheus/sample.conf:87]:
```suggestion
```

**Comment by powersj** [comment]
ok I think you are down to:

1. Run `make docs`
2. Sign the CLA

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @cnanfackgoto!

**Pull request merged**